csvGenerator.ipynb

```diff
@@ -0,0 +1,409 @@
+    "\n",
+    "        # Check if it's the first time writing to the file\n",
+    "        if flag == 0:\n",
+    "            fields = list(item.keys())\n",
+    "            writer.writerow(fields)\n",
+    "\n",
+    "        # Write the values to the CSV file\n",
+    "        writer.writerow(list(item.values()))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d5ec01d3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "The following function add_to_csv(csv_name: str, folder_name: str, items: list, csv_dict: dict, flag: int) -> int:\n",
+    "Add items to a dictionary object 'csv_dict' that contains all the collection items, with metadata and\n",
+    "downloaded file path.\n",
+    "\n",
+    "Parameters:\n",
+    "1. csv_name (str): Name of the CSV file.\n",
+    "2. folder_name (str): Name of the folder where the files will be downloaded.\n",
+    "3. items (list): List of collection items.\n",
+    "4. csv_dict (dict): Dictionary containing collection items with metadata and file paths.\n",
+    "5. flag (int): Flag to determine whether to write the fields at the top of the file.\n",
+    "\n",
+    "Returns:\n",
+    "int: Updated flag value.\n",
+    "\n",
+    "Functionality:\n",
+    "This function iterates through the collection items in 'items'. For each item, it checks if the item's key\n",
+    "already exists in 'csv_dict'. If not, it adds the item to 'csv_dict' with a unique key and sets the 'paths'\n",
+    "property to an empty string. If 'gilesUploads' are present in the item, it gets the set of file IDs and names,\n",
+    "downloads the files, and updates the 'paths' property accordingly. The function then writes the item to the\n",
+    "CSV file using the 'write_to_csv' function and updates the flag. Finally, it returns the updated flag value.\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "id": "37167759",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add the item to csv_dict that contains all the collection items (with metadata and downloaded file path)\n",
+    "def add_to_csv(csv_name: str, folder_name: str, items: list, csv_dict: dict, flag: int) -> int:\n",
+    "\n",
+    "    for item in items[\"items\"]:\n",
+    "        if item[\"key\"] in csv_dict:\n",
+    "            continue\n",
+    "        item[\"paths\"] = \"\"\n",
+    "\n",
+    "        # Check if gilesUploads are present\n",
+    "        if item[\"gilesUploads\"]:\n",
+    "            giles_ids = get_set(item[\"gilesUploads\"])\n",
+    "            if giles_ids:\n",
+    "                item[\"paths\"] = download_file(folder_name, giles_ids, auth_object.access_token)\n",
+    "\n",
+    "        # Add the item to csv_dict and write it to the CSV file\n",
+    "        csv_dict[item[\"key\"]] = item\n",
+    "        write_to_csv(csv_name, item, flag)\n",
+    "        flag = 1\n",
+    "\n",
+    "    return flag\n"
```

Comment:
what is the flag for? If it only needs to know if something was downloaded, why not return True/False? However, it doesn't look like you're using that info anywhere, so just remove it.